### PR TITLE
Make Broker/Routine/PIN Catalog as View

### PR DESCRIPTION
**Pull request by anshuldata**
Screenshot 2024-05-08 at 11 41 28 PM

Screenshot 2024-05-08 at 11 41 58 PM

Screenshot 2024-05-08 at 11 42 24 PM

Screenshot 2024-05-08 at 11 42 48 PM

### CODE REVIEW

File: bootstrap/021_external_functions.sql [hunk 1]
```diff
@@ -412,7 +372,7 @@ BEGIN
 
             create or replace external function internal.ef_run(unused object, request object)
             returns object
-            context_headers = (CURRENT_ACCOUNT, CURRENT_USER, CURRENT_ROLE, CURRENT_DATABASE, CURRENT_SCHEMA)
+            context_headers = (CURRENT_ACCOUNT, CURRENT_USER, CURRENT_ROLE, CURRENT_DATABASE, CURRENT_SCHEMA, CURRENT_REGION)
```
Comment (joshelser) [inline bootstrap/021_external_functions.sql:375]: Is this necessary for the `ef_run` endpoint to handle the request or are you just adding this for completeness?

File: bootstrap/021_external_functions.sql [hunk 2]
```diff
@@ -109,6 +109,17 @@ EXCEPTION
         RAISE;
 END;
 
+create or replace function internal.wrapper_ef_run(tenant_info object, request object)
```
Comment (joshelser) [inline bootstrap/021_external_functions.sql:112]: `tenant_info` appears to be unused by `ef_run`. Do we need to propagate that into this wrapper?

Instead, just make these calls `internal.ef_run(NULL, request)`

File: bootstrap/021_external_functions.sql (hunk 1 above)
Comment (anshuldata) [reply bootstrap/021_external_functions.sql:375]: yes, this is required for ef_run to work ([here](https://github.com/sundeck-io/sundeck/blob/f23ef6eeaa39234715275e4786094a5a97639ee9/services/query_service/pkg/udf_run_handler.go#L164-L171))

File: bootstrap/021_external_functions.sql (hunk 2 above)
Comment (anshuldata) [reply bootstrap/021_external_functions.sql]: fixed it

File: bootstrap/022_sundeck_views.sql [hunk 3]
```diff
@@ -1,31 +1,47 @@
...
+    hex_decode_string(b.VALUE:ROUTINE_NAME) as name,
+    hex_decode_string(b.VALUE:ROUTINE_NAMESPACE) as namespace,
+    hex_decode_string(b.VALUE:ROUTINE_REF_COUNT) as ref_count,
+    hex_decode_string(b.VALUE:ROUTINE_CREATED_AT) as created_at,
+    hex_decode_string(b.VALUE:ROUTINE_LAST_MODIFIED_AT) as last_modified_at,
+    hex_decode_string(b.VALUE:ROUTINE_DDL) as ddl,
+    hex_decode_string(b.VALUE:ROUTINE_COMMENT) as comment
+from b;
+
+create or replace view CATALOG.PINS as
```
Comment (pravindra) [inline bootstrap/022_sundeck_views.sql:35]: it's ok to add this in another PR. but, can you pls add grants, tokens, tables & parameters too ? For parity with "show sundeck xx"

File: bootstrap/044_brokers.sql [hunk 4]
```diff
@@ -1,17 +0,0 @@
-
```
Comment (pravindra) [inline bootstrap/044_brokers.sql:1]: pls create a follow-up jira to cleanup the code in the saas too ? maybe, after a week or so

**Review by pravindra — APPROVED** [review]

File: bootstrap/022_sundeck_views.sql (hunk 3 above)
Comment (anshuldata) [reply bootstrap/022_sundeck_views.sql:35]: Opened https://sundeck.atlassian.net/browse/SD-6842 to track it

File: bootstrap/044_brokers.sql (hunk 4 above)
Comment (anshuldata) [reply bootstrap/044_brokers.sql:1]: Opened https://sundeck.atlassian.net/browse/SD-6841 to track this

**Review by joshelser — APPROVED** [review]

**Pull request merged**